src/pages/NotFound.tsx

```diff
@@ -10,17 +13,34 @@ const NotFound = () => {
   }, [location.pathname]);
 
   return (
-    <div className="flex items-center justify-center bg-card-custom">
-      <div className="text-center">
-        <h1 className="text-4xl font-bold mb-4">404</h1>
-        <p className="text-xl text-gray-600 mb-4">Oops! Page not found</p>
-        <button
-          onClick={() => navigate('/')}
-          className="text-blue-500 hover:text-blue-700 underline"
-        >
-          Return to Home
-        </button>
-      </div>
+    <div className="flex min-h-screen w-full items-center justify-center bg-background p-4">
+      <Card className="w-full max-w-md shadow-lg">
+        <CardHeader className="text-center pb-2">
+          <div className="flex justify-center mb-4">
+            <div className="rounded-full bg-muted p-4">
+              <FolderX className="h-12 w-12 text-primary" />
+            </div>
+          </div>
+          <CardTitle className="text-4xl font-bold">404</CardTitle>
+          <CardDescription className="text-xl mt-2">
+            Página não encontrada
+          </CardDescription>
+        </CardHeader>
+        <CardContent className="flex flex-col items-center space-y-4 text-center">
+          <p className="text-muted-foreground">
+            A página que você está procurando não existe ou foi movida.
+          </p>
+          <p className="text-sm text-muted-foreground">
+            URL acessada: <code className="text-primary">{location.pathname}</code>
+          </p>
+          <Button 
+            onClick={() => navigate('/')}
+            className="mt-4 bg-primary hover:bg-primary/90"
+          >
+            Voltar para o início
+          </Button>
```

Comment:
Podemos utilizar o `Link` do `react-router-dom` aqui para fazer o redirecionamento.

```suggestion
          <Link to="/">
            Voltar para o início
          </Link>
```